detectors/aws/aws_test.go

```diff
@@ -53,32 +86,43 @@ func TestAWS_Detect(t *testing.T) {
 			},
 			Want: want{Error: "id not available"},
 		},
-		"Instance ID Available": {
+		"Hostname Not Found": {
 			Fields: fields{
-				Client: &clientMock{available: true, idDoc: func() (ec2metadata.EC2InstanceIdentityDocument, error) {
-					// Example from https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-identity-documents.html
-					doc := ec2metadata.EC2InstanceIdentityDocument{
-						MarketplaceProductCodes: []string{"1abc2defghijklm3nopqrs4tu"},
-						AvailabilityZone:        "us-west-2b",
-						PrivateIP:               "10.158.112.84",
-						Version:                 "2017-09-30",
-						Region:                  "us-west-2",
-						InstanceID:              "i-1234567890abcdef0",
-						InstanceType:            "t2.micro",
-						AccountID:               "123456789012",
-						PendingTime:             time.Date(2016, time.November, 19, 16, 32, 11, 0, time.UTC),
-						ImageID:                 "ami-5fb8c835",
-						Architecture:            "x86_64",
-					}
-
-					return doc, nil
-				}},
+				Client: &clientMock{available: true, idDoc: usWestInst, metadata: map[string]meta{}},
+			},
+			Want: want{Resource: resource.New(usWestIDLabels...)},
+		},
+		"Hostname Response Error": {
+			Fields: fields{
+				Client: &clientMock{
+					available: true,
+					idDoc:     usWestInst,
+					metadata: map[string]meta{
+						"hostname": {err: awserr.NewRequestFailure(awserr.New("EC2MetadataError", "failed to make EC2Metadata request", errors.New("response error")), http.StatusInternalServerError, "test-request")},
+					},
+				},
+			},
+			Want: want{Error: `partial resource: ["hostname": 500 EC2MetadataError]`, Partial: true, Resource: resource.New(usWestIDLabels...)},
+		},
+		"Hostname General Error": {
+			Fields: fields{
+				Client: &clientMock{available: true, idDoc: usWestInst, metadata: map[string]meta{"hostname": {err: errors.New("unknown error")}}},
+			},
+			Want: want{Error: `partial resource: ["hostname": unknown error]`, Partial: true, Resource: resource.New(usWestIDLabels...)},
+		},
+		"All Available": {
+			Fields: fields{
+				Client: &clientMock{available: true, idDoc: usWestInst, metadata: map[string]meta{"hostname": {value: "ip-12-34-56-78.us-west-2.compute.internal"}}},
```

Comment:
Total nitpick, but I'd be in favor of breaking up some of these long lines into multiple lines.

Reply:
I've split most cases of long lines now (struct fields), but I've left the AWS errors as single lines